### ISSUE TITLE
[Feature Request] command line parameter to adjust Max Pool Duration

### ISSUE THREAD

**Issue by atiannicelli** [osmlab/atlas-checks#417]
**Is your feature request related to a problem? Please describe.**
When running checks on global datasets we sometimes find that we are executing for more than 24 hours. the current pool duration limit is 24 hours and it quietly stops the spark job before it completes.

**Describe the solution you'd like**
It would be nice to be able to add a parameter on the command line like "-poolTimeoutMinutes=1440"

**Describe alternatives you've considered**
I can reduce the number of countries that we handle in a single run and increase the number of servers we use to process the checks.

**Additional context**
here is the spark-submit command I entered:
```
/opt/spark/bin/spark-submit --class=org.openstreetmap.atlas.checks.distributed.ShardedIntegrityChecksSparkJob --master=local[50] --conf='spark.driver.memory=400g' --conf='spark.rdd.compress=true' /home/ubuntu/atlas-checks/build/libs/atlas-checks-*-SNAPSHOT-shadow.jar -inputFolder='/home/ubuntu/atlas-dev/Atlas_Generation/test/daylight_0.5.0/atlas/' -output='/home/ubuntu/output/' -outputFormats='flags' -countries='ALB,AND,AUT,BLR,BEL,BIH,BGR,HRV,CYP,CHE,CZE,DEU,DNK,ESP,EST,FIN,FRA,GEO,GRC,GIB,HRV,HUN,IRL,ITA,LTU,LUX,LIE,LVA,MLT,MKD,NOR,NLD,POL,PRT,ROU,RUS,SVK,SRB,SVN,SWE,GBR,UKR' -configFiles='file:///home/ubuntu/atlas-dev/Atlas_Checks/utils/configurations/configuration.json'
```
and here is how it ended
```
20/11/11 18:28:28 INFO RunnableCheck: UnusualLayerTagsCheck completed in 1.678 minutes.
20/11/11 18:28:46 WARN Pool: Thread pool Countries Execution Pool has ended before it was terminated (maxDuration = 1440.000 minutes).
20/11/11 18:28:46 INFO ShardedIntegrityChecksSparkJob: Sharded checks completed in 1440.305 minutes
20/11/11 18:28:46 WARN FileSystemPerformanceHelper: Closing rename thread pool!
20/11/11 18:28:46 WARN FileSystemPerformanceHelper: Closed rename thread pool in 0 milliseconds!
20/11/11 18:28:46 INFO DynamicAtlasExpander: DynamicAtlas([11-1075-666]): Loading new shard 10-538-332 found a new Atlas ce09e0e1-9648-3478-8851-11a4f949f593.atlas of size [AtlasSize: edgeNumber=23093, nodeNumber=9366, areaNumber=17135, lineNumber=1329, pointNumber=3578, relationNumber=662]
20/11/11 18:28:47 INFO RunnableCheck: PoolSizeCheck completed in 2.050 minutes.
20/11/11 18:28:47 INFO Executor: Finished task 405.0 in stage 80.0 (TID 13730). 2141 bytes result sent to driver
20/11/11 18:28:47 INFO TaskSetManager: Starting task 456.0 in stage 80.0 (TID 13781, ip-172-31-12-41.us-west-1.compute.internal, executor driver, partition 456, PROCESS_LOCAL, 107663 bytes)
20/11/11 18:28:47 INFO TaskSetManager: Finished task 405.0 in stage 80.0 (TID 13730) in 795939 ms on ip-172-31-12-41.us-west-1.compute.internal (executor driver) (408/900)
20/11/11 18:28:47 INFO Executor: Running task 456.0 in stage 80.0 (TID 13781)
20/11/11 18:28:47 INFO SparkUI: Stopped Spark web UI at http://ip-172-31-12-41.us-west-1.compute.internal:4040
20/11/11 18:28:47 INFO DAGScheduler: Job 40 failed: Running checks on ITA, took 86402.077332 s
20/11/11 18:28:47 INFO DAGScheduler: ShuffleMapStage 76 (Running checks on PRT) failed in 86397.794 s due to Stage cancelled because SparkContext was shut down
20/11/11 18:28:47 INFO DAGScheduler: ShuffleMapStage 78 (Running checks on ITA) failed in 86397.702 s due to Stage cancelled because SparkContext was shut down
20/11/11 18:28:47 INFO DAGScheduler: ShuffleMapStage 46 (Running checks on HRV) failed in 86399.369 s due to Stage cancelled because SparkContext was shut down
20/11/11 18:28:47 INFO DAGScheduler: ShuffleMapStage 80 (Running checks on DEU) failed in 86397.602 s due to Stage cancelled because SparkContext was shut down
```

**Comment by atiannicelli** [comment]
As an update I have found that there is one check that is taking an extremely long time to finish when being executed against some specific shards. This is why BRA, NPL, etc keep coming up as skipped countries in our check runs. Even if we just specify a single country like NPL with all checks there is one or two checks that don't finish so the whole run is chalked up as a failure.

Here are the contents of the metrics files showing that the EdgeCrossingEdgeCheck took over 23 hours to complete on that one shard. in that same run we had another EdgeCrossingEdgeCheck against a BRA shard that did not complete in 24 hours so BRA ends up failing.
```
output/metric/NPL/11-1510-860_check-run-time-1605290097607.csv:EdgeCrossingEdgeCheck,1593062
output/metric/NPL/12-3019-1719_check-run-time-1605312296572.csv:EdgeCrossingEdgeCheck,23267419
output/metric/NPL/12-3018-1719_check-run-time-1605336179812.csv:EdgeCrossingEdgeCheck,47732627
output/metric/NPL/11-1509-860_check-run-time-1605372556841.csv:EdgeCrossingEdgeCheck,84074697
```

If there is a way to up the priority on this one it would be much appreciated.

**Comment by atiannicelli** [comment]
I created the following PR for this issue: https://github.com/osmlab/atlas-checks/pull/423

please let me know if that would be acceptable.

**Comment by Bentleysb** [comment]
https://github.com/osmlab/atlas-checks/pull/423 has been merged and released. 

I looked into the metrics for EdgeCrossingEdgeCheck on recent runs I have done. For the shards you mentioned, none took over 90 second to complete. There were a few shards for the check that took quite a while (around an hour). However none of those were in NPL or BRA, and LineCrossingWaterBodyCheck and InvalidMultipolygonRelationCheck were consistently the checks with the longest run times per shard, not EdgeCrossingEdgeCheck. So there is some optimization that can likely be done to EdgeCrossingEdgeCheck, but it does not appear that would have much of an effect on overall run time on my end. 

@atiannicelli, if you feel that there is still a major issue with EdgeCrossingEdgeCheck then we can keep this issue open and continue investigating. Otherwise, I think we are good to close this.
